google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAdCustomizer.java

```diff
@@ -0,0 +1,498 @@
+   * ad group.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupIds the ad group IDs to bind the feed items to.
+   * @param feedItemResourceNames the resource names of the feed items.
+   */
+  private void createFeedItemTargets(
+      GoogleAdsClient googleAdsClient, long customerId, List<Long> adGroupIds,
+      List<String> feedItemResourceNames) {
+
+    for(int i = 0; i < adGroupIds.size(); i++) {
+      Long adGroupId = adGroupIds.get(i);
+      String feedItemResourceName = feedItemResourceNames.get(i);
+
+      FeedItemTarget feedItemTarget = FeedItemTarget.newBuilder()
+          .setAdGroup(StringValue.of(ResourceNames.adGroup(customerId, adGroupId)))
+          .setFeedItem(StringValue.of(feedItemResourceName))
+          .build();
+
+      FeedItemTargetOperation feedItemTargetOperation = FeedItemTargetOperation.newBuilder()
+          .setCreate(feedItemTarget)
+          .build();
+
+      try (FeedItemTargetServiceClient feedItemTargetServiceClient =
+          googleAdsClient.getLatestVersion().createFeedItemTargetServiceClient()) {
+
+        MutateFeedItemTargetsResponse response =
+            feedItemTargetServiceClient.mutateFeedItemTargets(
+                Long.toString(customerId), ImmutableList.of(feedItemTargetOperation));
+
+        String feedItemTargetResourceName = response.getResults(0).getResourceName();
+        System.out.printf(
+            "Added feed item target with resource name %s.%n", feedItemTargetResourceName);
+      }
+    }
+  }
+
+  /**
+   * Creates expanded text ads that use the ad customizer feed to populate the placeholders.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupIds the ad group IDs to bind the feed items to.
+   * @param feedName the name of the feed.
+   */
+  private void createAdsWithCustomizations(
+      GoogleAdsClient googleAdsClient, long customerId, List<Long> adGroupIds, String feedName) {
+
+    // Creates an expanded text ad using the feed attribute names as placeholders.
+    ExpandedTextAdInfo expandedTextAdInfo = ExpandedTextAdInfo.newBuilder()
+        .setHeadlinePart1(StringValue.of(String.format("Luxury cruise to {=%s.Name}", feedName)))
+        .setHeadlinePart2(StringValue.of(String.format("Only {=%s.Price}", feedName)))
+        .setDescription(StringValue.of(String.format(
+            "Offer ends in {=countdown(%s.Date)}!", feedName)))
+        .build();
+
+    Ad ad = Ad.newBuilder()
+        .setExpandedTextAd(expandedTextAdInfo)
+        .addFinalUrls(StringValue.of("http://www.example.com"))
+        .build();
+
+    List<AdGroupAdOperation> adGroupAdOperations = new ArrayList<>();
+
+    // Creates the same ad in both ad groups. When they serve, they will show different values,
+    // since they match different feed items.
```

Comment:
A little more context would be helpful here to explain what's going on. This might be made easier to understand with a comment earlier per my previous comment.

Reply:
Added a comment in the previous section.